### PR TITLE
Checkout: add additional email verification notice

### PR DESCRIPTION
**Pull request by jhnstn**
#### Changes proposed in this Pull Request

* Add a new email verification warning for eCommerce plans

screen_shot_2018-12-02_at_8_49_14_pm

#### Testing instructions

* Create a new user and add them to the new plans test ( see the test plan for code-D21674 )
* Do not verify their email address
* Purchase a new eCommerce plan
* You should see a second notice indicating the need to verify the users email address

* Now verify the email address and try reloading the thank you page
* The notice should not be shown.

### CODE REVIEW

**Comment by obenland** [comment]
Should we disable the "Create your store!" button if it's unverified?

File: client/my-sites/checkout/checkout-thank-you/index.jsx [hunk 1]
```diff
@@ -379,9 +405,14 @@ export class CheckoutThankYou extends React.Component {
 			if ( delayedTransferPurchase ) {
 				planProps = {
 					action: (
-						<a className="thank-you-card__button" onClick={ this.startTransfer }>
+						<button
+							className="checkout-thank-you__button"
+							onClick={ this.startTransfer }
+							tabIndex={ 0 }
+							onKeyPress={ () => {} }
```
Comment (obenland) [inline client/my-sites/checkout/checkout-thank-you/index.jsx]: Should this also start the transfer?

Comment (jhnstn) [reply client/my-sites/checkout/checkout-thank-you/index.jsx]: I rolled back the attempt to update the for linting. I can circle back once we get this merged

**Comment by jhnstn** [comment]
> Should we disable the "Create your store!" button if it's unverified?

That was considered, we are leaving it enabled for now since we want knowledgeable customers to verify their email in a different tab. cc @sararosso

**Pull request merged**